packages/core/src/shared/telemetry/telemetryClient.ts

```diff
@@ -108,7 +108,7 @@ export class DefaultTelemetryClient implements TelemetryClient {
                         ClientID: this.clientId,
                         OS: os.platform(),
                         OSVersion: os.release(),
-                        ComputeEnv: getComputeEnvType(),
+                        ComputeEnv: await getComputeEnvType(),
```

Comment:
Just noticed that this is called by `postMetrics`, not just once per session. Should `getComputeEnvType` be memoized?